.github/workflows/deploy.yml

```diff
@@ -0,0 +1,40 @@
+name: Deploy
+
+permissions: read-all
+
+on:
+  workflow_dispatch:
+    inputs:
+      plugin-directory:
+        description: "Plugin directory"
+        required: true
+
+jobs:
+  deploy-plugin:
+    runs-on: ubuntu-latest
+    name: Deploy plugin to BSR
+    timeout-minutes: 10
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.19"
+      - name: Install buf cli
+        run: |
+          go install github.com/bufbuild/buf/cmd/buf@main
```

Comment:
We didn't release the buf cli with the required plugin push command, so installing from source until next release.

Also didn't use `bufbuild/buf:latest` because passing secrets to docker isn't ideal.